### PR TITLE
StreamUtils: new util class with .toUnmodifiableList()

### PR DESCRIPTION
**Pull request by msgilligan**
This will also make it easier to find and upgrade these methods when we upgrade to newer JDKs.

I'm certain there are other methods we will want to add to this utility class.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/base/utils/StreamUtils.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package org.bitcoinj.base.utils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Stream Utilities. bitcoinj is moving towards functional-style programming, immutable data structures, and
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/base/utils/StreamUtils.java:26]: Nitpick: I think when the word "bitcoinj" is used as first word of a sentence it should read "Bitcoinj", like any other word.

**Comment by schildbach** [comment]
Looks good!

File: core/src/main/java/org/bitcoinj/base/utils/StreamUtils.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/base/utils/StreamUtils.java]: Fixed and force-pushed.

**Comment by schildbach** [comment]
Removed some (now unused) imports and merged.

**Pull request closed without merging**